packages/api/src/router/comment.ts

```diff
@@ -0,0 +1,123 @@
+import { Prisma } from "@prisma/client";
+import { TRPCError } from "@trpc/server";
+import { z } from "zod";
+
+import { createTRPCRouter, protectedProcedure } from "../trpc";
+
+export const commentRouter = createTRPCRouter({
+  list: protectedProcedure
+    .input(
+      z.object({
+        postId: z.number().nonnegative(),
+      }),
+    )
+    .query(async ({ input, ctx }) => {
+      try {
+        return await ctx.db.comment.findMany({
+          where: { postId: input.postId },
+          orderBy: { createdAt: "desc" },
+          select: {
+            id: true,
+            text: true,
+            createdAt: true,
+          },
+        });
+      } catch (e) {
+        if (e instanceof Prisma.PrismaClientKnownRequestError) {
+          if (e.code === "P2025") {
+            throw new TRPCError({
+              code: "NOT_FOUND",
+              message: "Post not found",
+            });
+          }
+        }
+        throw e;
+      }
+    }),
+  create: protectedProcedure
+    .input(
+      z.object({
+        postId: z.number().nonnegative(),
+        text: z.string(),
+      }),
+    )
+    .mutation(async ({ input, ctx }) => {
+      try {
+        const userId = ctx.userId;
+        return await ctx.db.comment.create({
+          data: {
+            text: input.text,
+            post: { connect: { id: input.postId } },
+            user: { connect: { id: userId } },
+          },
+        });
+      } catch (e) {
+        if (e instanceof Prisma.PrismaClientKnownRequestError) {
+          if (e.code === "P2025") {
+            throw new TRPCError({
+              code: "NOT_FOUND",
+              message: "Post not found",
+            });
+          }
+        }
+        throw e;
+      }
+    }),
+  update: protectedProcedure
+    .input(
+      z.object({
+        id: z.number().nonnegative(),
+        text: z.string(),
+      }),
+    )
+    .mutation(async ({ input, ctx }) => {
+      const userId = ctx.userId;
+      const comment = await ctx.db.comment.findUnique({
+        where: { id: input.id },
+        select: { id: true, userId: true },
+      });
```

Comment:
will this throw an error if the comment doesn't exist?

Reply:
yeah right after i check `if (!comment)`

Reply:
i mean like it might throw a prisma error?

Reply:
no, it just returns null:
<img width="393" alt="image" src="https://github.com/uoftblueprint/centre-for-dreams/assets/40612523/43fad0a8-eb39-4622-bdc0-886ebf45615c">
a prisma error is only thrown when connecting to an entity that doesn't exist

Reply:
also i changed error handling for `comment.list` so that it correctly throws 404 when the provided `postId` does not exist (before it would just return an empty list)